mypyc/genops.py

```diff
@@ -3495,6 +3594,44 @@ def instantiate_generator_class(self) -> Value:
         self.add(SetAttr(curr_env_reg, NEXT_LABEL_ATTR_NAME, zero_reg, fitem.line))
         return generator_reg
 
+    def add_raise_exception_blocks_to_generator_class(self, line: int) -> None:
+        """
+        Generates blocks to check if error flags are set while calling the helper method for
+        generator functions, and raises an exception if those flags are set.
+        """
+        cls = self.fn_info.generator_class
+
+        # Check to see if an exception was raised.
+        error_block = BasicBlock()
+        ok_block = BasicBlock()
+        comparison = self.binary_op(cls.exc_type_reg, self.none(), 'is not', line)
+        self.add_bool_branch(comparison, error_block, ok_block)
+
+        self.activate_block(error_block)
+
+        # Check to see if the traceback field is set.
```

Comment:
Cut this logic and just always raise with a traceback. If that doesn't actually work, I'd move the conditional logic into the helper for it.